File: typescript/core/medium/student_who_will_replace_chalk.ts
function chalkReplacer(chalk: number[], k: number): number {
  let student = 0
  const sumOfCircle = chalk.reduce((sum, curr) => sum + curr, 0)
  k = k % sumOfCircle
  for (let i = 0; i < chalk.length; i++) {
    if (k < chalk[i]) return i
    k -= chalk[i]
  }
  return student
}

console.log(chalkReplacer([5,1,5], 22))

File: golang/core/medium/student_who_will_replace_chalk/student_who_will_replace_chalk.go
package main

import (
	"fmt"
)

func chalkReplacer(chalk []int, k int) int {
	sumOfCircle := 0
	for _, c := range chalk {
		sumOfCircle += c
	}
	k %= sumOfCircle
	for i, c := range chalk {
		if k < c {
			return i
		}
		k -= c
	}
  return 0
}

func main() {
	fmt.Println(chalkReplacer([]int{5, 1, 5}, 22))
}